### PR TITLE
Issue 2772 : Smoke tests crash with OpenMP

### LINKED ISSUE #2772: Smoke tests crash with OpenMP
Running our smoke tests with OMP enabled results in a memory error.

This can be seen on OSX, and our CircleCI tests that run on Ubuntu.

### PR DESCRIPTION
**Pull request by clyne**
Fixes #2772 

This PR cleans up some memory use and hopefully addresses 2772. However, since I can't reproduce the problem it will need to be tested by someone who can. CC @sgpearse

### CODE REVIEW

File: lib/vdc/StretchedGrid.cpp [hunk 1]
```diff
@@ -320,13 +320,10 @@ float StretchedGrid::GetValueLinear(const CoordType &coords) const
...
-    auto ndims = StructuredGrid::GetNumDimensions();
 
-    DimsType min, max;
-    for (int i = 0; i < ndims; i++) {
-        min[i] = 0;
-        max[i] = (dims[i] - 1);
-    }
+    DimsType min = {0, 0, 0};
+    DimsType max = {0, 0, 0};
+    for (int i = 0; i < dims.size(); i++) { max[i] = (dims[i] - 1); }
```
Comment (shaomeng) [inline lib/vdc/StretchedGrid.cpp:326]:
```suggestion
    for (int i = 0; i < dims.size(); i++) { 
        assert( dims[i] > 0 ); // will help debug
        max[i] = (dims[i] - 1); 
    }
```

**Review by shaomeng — COMMENTED** [review]
An assert that will help debug bad values.

**Comment by clyne** [comment]
@shaomeng just pinging you that I need a final review for this PR. Thanks

**Review by shaomeng — APPROVED** [review]
Sorry forgot to look at this. It's all good now!

**Pull request merged**